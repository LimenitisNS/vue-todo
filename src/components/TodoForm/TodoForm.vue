<template>
  <el-form v-model="form" class="todo-form">
    <el-form-item prop="value">
      <el-input v-model="form.value" class="todo-form__input" />
    </el-form-item>
    <el-form-item>
      <el-button class="todo-form__button" @click="handleCreateTodo"
        >Создать</el-button
      >
    </el-form-item>
  </el-form>
</template>

<script lang="ts">
import { defineComponent, reactive, toRefs } from "vue";
import { useStore } from "vuex";

export default defineComponent({
  name: "TodoForm",
  setup() {
    const store = useStore();

    const data = reactive({
      form: {
        value: "",
      },
    });

    const handleCreateTodo = (): void => {
      store.dispatch("addTodo", data.form.value);
      data.form.value = "";
    };

    return {
      ...toRefs(data),

      handleCreateTodo,
    };
  },
});
</script>
