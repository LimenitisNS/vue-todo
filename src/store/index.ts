import Vuex, { StoreOptions } from "vuex";
import todo from "./modules/todo";
import { RootState } from "@/store/type";

const store: StoreOptions<RootState> = {
  modules: {
    todo: todo,
  },
};

export default new Vuex.Store<RootState>(store);
