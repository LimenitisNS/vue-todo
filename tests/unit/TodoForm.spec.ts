import { mount } from "@vue/test-utils";
import TodoForm from "@/components/TodoForm/TodoForm.vue";
import store from "@/store";

describe("TodoForm", () => {
  it("snapshot", () => {
    const wrapper = mount(TodoForm);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("create todo", async () => {
    const value = "test";

    const wrapper = mount(TodoForm, {
      global: {
        plugins: [store],
      },
    });

    const input = await wrapper.find(".todo-form__input");
    const button = await wrapper.find(".todo-form__button");

    (input.element as HTMLInputElement).value = value;
    input.trigger("input");

    expect((input.element as HTMLInputElement).value).toBe(value);

    await wrapper.vm.$nextTick();

    button.trigger("click");

    await wrapper.vm.$nextTick();

    expect(store.getters["list"].length).toBe(1);
  });
});
