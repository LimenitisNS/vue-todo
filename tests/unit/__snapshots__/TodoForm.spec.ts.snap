// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm snapshot 1`] = `
<el-form
  class="todo-form"
  modelvalue="[object Object]"
>
  <el-form-item
    prop="value"
  >
    <el-input
      class="todo-form__input"
      modelvalue=""
    />
  </el-form-item>
  <el-form-item>
    <el-button
      class="todo-form__button"
    >
      Создать
    </el-button>
  </el-form-item>
</el-form>
`;
